src/ai/susi/mind/SusiSkill.java

```diff
@@ -547,65 +567,60 @@ public static JSONObject getSkillMetadata(String model, String group, String lan
 
                             skillName.put("stars", newFiveStarRating);
                         }
-                        skillMetadata.put("skill_rating", skillName);
-                    }
-                    else {
-                        JSONObject newRating=new JSONObject();
-                        newRating.put("negative", "0");
-                        newRating.put("positive", "0");
-                        newRating.put("feedback_count", 0);
-
-                        JSONObject newFiveStarRating=new JSONObject();
-                        newFiveStarRating.put("one_star", 0);
-                        newFiveStarRating.put("two_star", 0);
-                        newFiveStarRating.put("three_star", 0);
-                        newFiveStarRating.put("four_star", 0);
-                        newFiveStarRating.put("five_star", 0);
-                        newFiveStarRating.put("avg_star", 0);
-                        newFiveStarRating.put("total_star", 0);
-
-                        newRating.put("stars", newFiveStarRating);
-
-                        skillMetadata.put("skill_rating", newRating);
+                        return skillName;
                     }
                 }
             }
-            else {
-                JSONObject newRating=new JSONObject();
-                newRating.put("negative", "0");
-                newRating.put("positive", "0");
-                newRating.put("feedback_count", 0);
-
-                JSONObject newFiveStarRating=new JSONObject();
-                newFiveStarRating.put("one_star", 0);
-                newFiveStarRating.put("two_star", 0);
-                newFiveStarRating.put("three_star", 0);
-                newFiveStarRating.put("four_star", 0);
-                newFiveStarRating.put("five_star", 0);
-                newFiveStarRating.put("avg_star", 0);
-                newFiveStarRating.put("total_star", 0);
-
-                newRating.put("stars", newFiveStarRating);
-
-                skillMetadata.put("skill_rating", newRating);
-            }
         }
+        JSONObject newRating=new JSONObject();
+        newRating.put("negative", "0");
+        newRating.put("positive", "0");
+        newRating.put("feedback_count", 0);
+
+        JSONObject newFiveStarRating=new JSONObject();
+        newFiveStarRating.put("one_star", 0);
```

Comment:
very strange data structure. Who created this data structure by naming of the stars? You could have used an JSONArray containing integers for the five stars.